front/src/components/boxs/device-in-room/device-features/sensor-value/BadgeNumberDeviceValue.jsx

```diff
@@ -79,15 +80,28 @@ const BADGE_VALUE_CONVERTERS = {
 };
 
 const BadgeNumberDeviceValue = props => {
-  const { category, type, last_value: lastValue = null, unit } = props.deviceFeature;
+  const { user, deviceFeature } = props;
+  const { category, type, last_value: lastValue = null, unit } = deviceFeature;
+
+  const userUnit = user.distance_unit_preference;
 
   const colorMethod = BADGE_CATEGORIES[category];
   if (!colorMethod) {
     return <RawDeviceValue {...props} />;
   }
 
-  let value = lastValue === null ? -1 : lastValue;
   let valueIsEnum = false;
+
+  let value = lastValue;
+  let displayUnit = unit;
+
+  const isDistanceUnit = DISTANCE_UNITS.DEVICE_FEATURE_UNITS.includes(displayUnit);
```

Comment:
Just raising a thought: what if we split this into a dedicated component?

That’s what we did for the Temperature features, for example.

I’m a bit concerned that we’re adding distance-specific logic into a generic file that handles many types of devices. As we introduce more conversions or device types, it might become harder to maintain and reason about.

What’s your take on this?

Reply:
Hello @Pierre-Gilles ,
I agree with your suggestion. I just thought it would be more cumbersome, especially if I added the volume, area, elevation, distance, and speed conversions later, but your solution actually seems much more readable.

The change is complete.

However, is there still the modification made to the `Gauge` component? As well as the one in the `Chart` component, which I'm currently adding...? What do you think?